src/screens/Vault/SignerAdvanceSettings.tsx

```diff
@@ -1410,3 +1451,23 @@ const styles = StyleSheet.create({
     marginBottom: hp(15),
   },
 });
+
+function matchSigners(signersList, signers) {
+  // signersList-psbt | signers-vault
+  const isMatchingObject = (psbtSigner, vaultSigner) => {
+    const generatePub = getPubKeyFromXpub(vaultSigner.xpub, psbtSigner.derivationPath);
+    return psbtSigner.pubKey === generatePub;
+  };
+
+  for (const psbtSigner of signersList) {
+    const match = signers.find(
```

Comment:
We want to check all vault signers not just the signer we use right now. We need to ensure all the signers used are correct.

Also need to ensure the derivations up to the hardened part match the derivations of the keys in the saved vault.